src/bindings/build.py

```diff
@@ -34,6 +34,17 @@
     with open(header, "r") as hfile:
         ffi.cdef(hfile.read())
 
+# we need to easily control a few values for Windows builds
+source = []
+if os.getenv("PYNACL_SODIUM_STATIC") is not None:
+    source.append("#define SODIUM_STATIC")
+
+source.append("#include <sodium.h>")
+
+if os.getenv("PYNACL_SODIUM_LIBRARY_NAME") is not None:
+    libraries = [os.getenv("PYNACL_SODIUM_LIBRARY_NAME")]
```

Comment:
Why wouldn't it be named that -- is there a different default for
self-compiled, no default, or just "someone might change it"?

On Sat, Jan 9, 2016 at 10:53 AM, Paul Kehrer <notifications@github.com>
wrote:

> In src/bindings/build.py
> <https://github.com/pyca/pynacl/pull/164#discussion_r49265121>:
>
> > @@ -34,6 +34,17 @@
> >      with open(header, "r") as hfile:
> >          ffi.cdef(hfile.read())
> >
> > +# we need to easily control a few values for Windows builds
> > +source = []
> > +if os.getenv("PYNACL_SODIUM_STATIC") is not None:
> > +    source.append("#define SODIUM_STATIC")
> > +
> > +source.append("#include <sodium.h>")
> > +
> > +if os.getenv("PYNACL_SODIUM_LIBRARY_NAME") is not None:
> > +    libraries = [os.getenv("PYNACL_SODIUM_LIBRARY_NAME")]
>
> If you compile it yourself it won't necessarily be named that, but as I
> said before, if anyone is compiling it themselves it is perhaps reasonable
> to state that they need to learn what the expected name is for linking.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/pynacl/pull/164/files#r49265121>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084